File: algorithms/169. Majority Element/solution.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        equal = 0
        i = 0
        j = len(nums) - 1
        while i < j:
            if nums[i] != nums[j]:
                if equal > 0:
                    equal -= 1
                else:
                    j -= 1
            else:
                equal += 1
            i += 1
        return nums[j]

File: algorithms/169. Majority Element/solution.cc
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> counter;
        int m = nums.size() / 2;
        for ( int n : nums ) {
            if ( ++counter[n] > m ) {
                return n;
            }
        }
    }
};

// 44 / 44 test cases passed.
// Status: Accepted
// Runtime: 33 ms
